staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/compilation.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/checker/decls"
+	expr "google.golang.org/genproto/googleapis/api/expr/v1alpha1"
+	"google.golang.org/protobuf/proto"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	"k8s.io/apiextensions-apiserver/pkg/apiserver/schema"
+	celmodel "k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model"
+)
+
+// ScopedTypeName is the placeholder type name used for the type of ScopedVarName if it is an object type.
+const ScopedTypeName = "apiextensions.k8s.io.v1alpha1.ValidationExpressionSelf"
+
+// ScopedVarName is the variable name assigned to the locally scoped data element of a CEL valid.
+const ScopedVarName = "self"
+
+// CompilationResult represents the cel compilation result for one rule
+type CompilationResult struct {
+	Rule      apiextensions.ValidationRule
+	Program   cel.Program
+	Error     *Error
+	RuleIndex int
```

Comment:
I think you don't really need `RuleIndex` or `Rule`, as the compilation results have the same order as `schema.XValidations` in `validation.go`

Reply:
Simplified CompilationResult